src/SDK/ConfigBuilder.php

```diff
@@ -0,0 +1,199 @@
+                        'schedule_delay' => Expect::string()->castTo('int'),
+                        'export_timeout' => Expect::string()->castTo('int'),
+                        'max_queue_size' => Expect::string()->castTo('int'),
+                        'max_export_batch_size' => Expect::string()->castTo('int'),
+                    ]),
+                    'simple' => Expect::structure([]),
+                    'noop' => Expect::structure([]),
+                ]),
+                'exporter' => Expect::string('grpc'),
+                'exporters' => Expect::structure([
+                    'zipkin' => Expect::structure([
+                        'endpoint' => Expect::string(),
+                        'timeout' => Expect::string()->castTo('int'),
+                    ]),
+                    'otlp' => Expect::structure([
+                        'endpoint' => Expect::string(),
+                        'insecure' => Expect::anyOf('true', 'false', '0', '1'),
+                        'certificate_file' => Expect::string(),
+                        'headers' => Expect::string(),
+                        'compression' => Expect::string(),
+                        'timeout' => Expect::string()->castTo('int'),
+                        'protocol' => Expect::string('http/protobuf'),
+                        
+                    ]),
+                    'jaeger' => Expect::structure([
+                        'agent_host' => Expect::string(),
+                        'agent_port' => Expect::string()->castTo('int'),
+                        'endpoint' => Expect::string(),
+                        'timeout' => Expect::string()->castTo('int'),
+                        'user' => Expect::string(),
+                        'password' => Expect::string(),
+                    ]),
+                ]),
+            ]),
+            'span' => Expect::structure([
+                'limits' => Expect::structure([
+                    'attribute_count' => Expect::string()->castTo('int'),
+                    'attribute_value_length' => Expect::string()->castTo('int'),
+                    'event_count' => Expect::string()->castTo('int'),
+                    'link_count' => Expect::string()->castTo('int'),
+                    'attribute_per_event' => Expect::string()->castTo('int'),
+                    'attribute_per_link' => Expect::string()->castTo('int'),
+                ]),
+            ]),
+        ]);
+        $env = $this->env();
+        $resourceAttributes = $this->resourceAttributes();
+
+        return (new Processor())->processMultiple($schema, [$env, $resourceAttributes, $data]);
+    }
+
+    /**
+     * Retrieve all OTEL_* environment variables, and map them to a nested array
+     */
+    private function env(): array
+    {
+        $vars = array_filter(getenv(), function ($key) {
+            return strpos($key, 'OTEL_') !== false;
+        }, ARRAY_FILTER_USE_KEY);
+        $output = [];
+        foreach ($vars as $key => $val) {
```

Comment:
A long time ago I read a [book on code readebilty](https://www.oreilly.com/library/view/the-art-of/9781449318482/), so you can blame the authors for me complaininga bout this stuff. :)

 However while a lot of things there seam to be nit-picky at first, they make sense, when one thinks about and get used to them. One point is to not use "short variable names". While you and maybe me as well know what's meant with "$val", it might not be obviouss to everyone.